<a href="https://colab.research.google.com/github/SoniaVG/Analysis-of-data/blob/main/Titanic_Inferencial_Statistics.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Inferential Statistics




In [ ]:
! pip install -q kaggle
from google.colab import files
files.upload()

Saving kaggle.json to kaggle.json


{'kaggle.json': b'{"username":"soniavenancioguzmn","key":"8f7c58a031531fdfbf6e2d5d2e98bbe6"}'}

In [ ]:
! mkdir ~/.kaggle
! cp kaggle.json ~/.kaggle/
! chmod 600 ~/.kaggle/kaggle.json

#**Data Dictionary**

$$\tiny{ \begin{matrix}
\text{Variable} &	\text{Definition}	& \text{Key} \\ \hline
survival & 	Survival 	&  0 = No, 1 = Yes \\ \hline
pclass 	& \text{Ticket class} &	1 = 1st, 2 = 2nd, 3 = 3rd \\ \hline
sex 	& Sex 	& \\ \hline
Age 	& \text{Age in years} 	& \\ \hline
sibsp  & \text{of siblings / spouses aboard the Titanic} 	& \\ \hline
parch 	& \text{of parents / children aboard the Titanic} & \\ \hline
ticket 	& \text{Ticket number }&	\\ \hline
fare 	& \text{Passenger fare} & 	\\ \hline
cabin &	\text{Cabin number} &	\\ \hline
embarked &	\text{Port of Embarkation} &	\text{C = Cherbourg, Q = Queenstown, S = Southampton}
\end{matrix}
}
$$

In [ ]:
!kaggle datasets download -d cities/titanic123
!unzip "/content/titanic123.zip"
import pandas as pd
import numpy as np
df=pd.read_csv("ChanDarren_RaiTaran_Lab2a.csv")
df.head(3)

  0% 0.00/22.1k [00:00<?, ?B/s]
100% 22.1k/22.1k [00:00<00:00, 29.5MB/s]
Archive:  /content/titanic123.zip
  inflating: ChanDarren_RaiTaran_Lab2a.csv  


PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0            1         0       3  ...   7.2500   NaN         S
1            2         1       1  ...  71.2833   C85         C
2            3         1       3  ...   7.9250   NaN         S

[3 rows x 12 columns]

In [ ]:
colnames=['PassengerId', 'Survived', 'Pclass', 'Sex', 'Age', 'SibSp','Parch']
df_final=df[colnames]
df_final= df_final.fillna(df_final["Age"].mean())
df_final.head(2)

PassengerId  Survived  Pclass     Sex   Age  SibSp  Parch
0            1         0       3    male  22.0      1      0
1            2         1       1  female  38.0      1      0

# Will surviving be independent of gender? 

>1.  **Chi-square test**: The relationship between categorical variables
>2.  **Remark**: The chi-square test cannot show the strength of the correlation between two categorical variables, but it can only show whether the two categorical variables are related. 

$$
\begin{eqnarray}
H_0&:& \text{ the gender and survival variables are independent }\\
H_a&:& \text{ there is a relationship between the variables }
\end{eqnarray}
$$






In [ ]:
#from pandas.core.frame import DataFrame
#GENDER / SURVIVOR contingency table 
table1=pd.crosstab(index= df_final['Survived'],columns=df_final['Sex'], margins=True)
table1

Sex       female  male  All
Survived                   
0             81   468  549
1            233   109  342
All          314   577  891

In [ ]:
#tabla de frecuencia real
table_last = table1.div(table1['All'], axis=0)
print(table_last)

Sex         female      male  All
Survived                         
0         0.147541  0.852459  1.0
1         0.681287  0.318713  1.0
All       0.352413  0.647587  1.0


In [ ]:
#frecuencia esperada
from scipy import stats
print('chisq = %6.4f\n p-value = %6.4f\n dof = %i\n table_expected_freq =\n %s' %stats.chi2_contingency(table1))

chisq = 263.0506
 p-value = 0.0000
 dof = 4
 table_expected_freq =
 [[193.47474747 355.52525253 549.        ]
 [120.52525253 221.47474747 342.        ]
 [314.         577.         891.        ]]


**Conclusion:**
Considerando que el valor de estadístico de prueba $x^2=263$ (cae en la zona de rechazo), $\alpha=0.05 > p_{value}=0$ entonces  rechazar $H_0$, en consecuencia, si existe una relación entre haber sobrevivido y ser hombre o mujer.




# Will surviving be independent of class? 

$$
\begin{eqnarray}
H_0 &:& \text{ the class and survival variables are independent }\\
H_a &:& \text{ there is a relationship between the variables }
\end{eqnarray}
$$


In [ ]:
table2=pd.crosstab(index= df_final['Survived'],columns=df_final['Pclass'], margins=True)
table2

Pclass      1    2    3  All
Survived                    
0          80   97  372  549
1         136   87  119  342
All       216  184  491  891

In [ ]:
print('chisq = %6.4f\n p-value = %f\n dof = %i\n table_expected_freq =\n %s' %stats.chi2_contingency(table2))


chisq = 102.8890
 p-value = 0.000000
 dof = 6
 table_expected_freq =
 [[133.09090909 113.37373737 302.53535354 549.        ]
 [ 82.90909091  70.62626263 188.46464646 342.        ]
 [216.         184.         491.         891.        ]]


#Conclusion
Sabemos que si: $p_{value}=0<0.05=\alpha$ entonces, rechazar $H_0$. Por tanto, existe una relación entre haber sobrevivido y pertenecer a alguna de las clases.